### ISSUE TITLE
MatrixBandPart op in tf2onnx does not support bool inputs

### ISSUE THREAD

**Issue by Gerstenberger** [onnx/tensorflow-onnx#2268]
**Describe the bug**

For the op `MatrixBandPart`, inputs of type `bool` are not supported by `tf2onnx`. The corresponding op in TensorFlow does support bool types. The following error appears when converting `tf.linalg.band_part` with boolean input
```
onnxruntime.capi.onnxruntime_pybind11_state.InvalidGraph: [ONNXRuntimeError] : 10 : INVALID_GRAPH : Load model from model.onnx failed:This is an invalid model.
Type Error: Type 'tensor(bool)' of input parameter (Cast__43:0) of operator (Mul) in node (PartitionedCall/MatrixBandPart) is invalid.
```

**Urgency**

Appears in a couple of Transformer language models. We can provide a simple fix in our TF code in the mean time. Not so urgent.

**System information**

- OS Platform and Distribution (e.g., Linux Ubuntu 18.04*): Linux Ubuntu 18.04.6 LTS
- TensorFlow Version: 2.14.0, 2.13.1, 2.9.0 tested so far
- Python version: 3.9
- ONNX version (if applicable, e.g. 1.11*): 1.15.0
- ONNXRuntime version (if applicable, e.g. 1.11*): 1.16.1

**To Reproduce**
```python
import tensorflow as tf

class CustomModule(tf.Module):

    @tf.function(input_signature=[tf.TensorSpec([None, None], tf.bool)])
    def __call__(self, x):
        return tf.linalg.band_part(x, -1, 0)

bandpart = CustomModule()

x = tf.ones([3, 3], dtype=tf.bool)
print(bandpart(x))

tf.saved_model.save(bandpart, "saved-model-example")
```
then
```
python3 -m tf2onnx.convert --saved-model saved-model-example --output model.onnx
```
and then running
```python
import numpy as np
import onnxruntime as ort

ort_sess = ort.InferenceSession('model.onnx')
x = np.ones([3, 3], dtype=bool)

print(ort_sess.run(None, {'x': x}))

```

**Screenshots**

None

**Additional context**

We use this op with boolean inputs to create a attention mask.

One workaround would be
```python
def onnx_compat_matrix_band_part(x, num_lower, num_upper):
    if x.dtype == tf.bool:
        inputs = tf.cast(x, dtype=tf.float32)
    else:
        inputs = x                                                                                                                                                                                        
    res = tf.linalg.band_part(inputs, num_lower, num_upper)
    if x.dtype == tf.bool:
        return tf.cast(res, dtype=tf.bool)
    return res
```
See issue https://github.com/rwth-i6/returnn/issues/1455 for some notes.
